generate-windows-versions.rb

```diff
@@ -31,18 +31,62 @@
   min_requirements.any? { |req| req.satisfied_by?(Gem::Version.new(version)) }
 }.to_h
 
+latest_version = Gem::Version.new(versions.keys.last)
+head_version = Gem::Version.new("#{latest_version.bump}.0dev")
+
 versions['head'] = {
-  'arm64': 'https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-head/rubyinstaller-head-arm.7z',
-  'x64': 'https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-head/rubyinstaller-head-x64.7z'
+  'arm64' => 'https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-head/rubyinstaller-head-arm.7z',
+  'x64' => 'https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-head/rubyinstaller-head-x64.7z'
 }
 versions['mingw'] = {
-  'x64': 'https://github.com/MSP-Greg/ruby-loco/releases/download/ruby-master/ruby-mingw.7z'
+  'x64' => 'https://github.com/MSP-Greg/ruby-loco/releases/download/ruby-master/ruby-mingw.7z'
 }
 versions['mswin'] = {
-  'x64': 'https://github.com/MSP-Greg/ruby-loco/releases/download/ruby-master/ruby-mswin.7z'
+  'x64' => 'https://github.com/MSP-Greg/ruby-loco/releases/download/ruby-master/ruby-mswin.7z'
 }
 versions['ucrt'] = {
-  'x64': 'https://github.com/MSP-Greg/ruby-loco/releases/download/ruby-master/ruby-ucrt.7z'
+  'x64' => 'https://github.com/MSP-Greg/ruby-loco/releases/download/ruby-master/ruby-ucrt.7z'
 }
 
 File.binwrite 'windows-versions.json', "#{JSON.pretty_generate(versions)}\n"
+
+base_url = 'https://github.com/ntkme/setup-msys2-gcc/releases/latest/download/windows-toolchain.json'
+windows_toolchain = JSON.parse(URI.open(base_url) { |f| f.read }, symbolize_names: true)
+
+versions.each do |raw_version, archs|
+  version = Gem::Version.correct?(raw_version) ? Gem::Version.new(raw_version) : head_version
+
+  archs.each_key do |raw_arch|
+    arch = raw_arch == 'arm64' ? 'aarch64' : raw_arch
+
+    platform = case raw_version
+               when 'head', 'ucrt'
```

Comment:
```suggestion
    when 'head', 'ucrt'
```
Could you align as if there was no assignment (IMO assignment should never cause indentation), or if your editor gets in the way then a single indent after `=`:
```
platform =
  case
  when
```
?